CartingService/Messaging/MessagingService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Text.Json;
+using Confluent.Kafka;
+using Microsoft.Extensions.Logging;
+
+public class MessagingService
+{
+    private readonly IItemService _itemService;
+    private readonly ILogger _logger;
+
+    public MessagingService(IItemService itemService, ILogger<MessagingService> logger)
+    {
+        _itemService = itemService;
+        _logger = logger;
+    }
+
+    public async Task Consume(string server, string groupId, CancellationToken cancellationToken)
+    {
+        using var consumer = new ConsumerBuilder<Ignore, string>(new ConsumerConfig { GroupId = groupId, BootstrapServers = server }).Build();
+
+        consumer.Subscribe(nameof(ItemUpdatedMessage));
+        while (!cancellationToken.IsCancellationRequested)
+        {
+            var message = consumer.Consume(cancellationToken);
+            var messageHandled = false;
```

Comment:
How do you handle the scenario if the app has suddenly stopped at this moment of execution? Will Kafka try to deliver the message again? or will it be lost?

Reply:
The `Commit` call further in the function takes care of acknowledging the message